soundmeter/meter.py

```diff
@@ -124,27 +132,64 @@ def start(self):
                 record.send(True)  # Record stream `AUDIO_SEGMENT_LENGTH' long
                 data = self.output.getvalue()
                 segment = pydub.AudioSegment(data)
-                rms = segment.rms
+                if self.valuestype == "rms":
+                    soundvalue = segment.rms
+                elif self.valuestype == "db":
+                    soundvalue = self.weightedvalue(segment, self.weightingcurve)
+                else:
+                    sys.exit(1)  # Value type must be either db or rms
                 if self.collect:
-                    self.collect_rms(rms)
-                self.meter(rms)
+                    self.collect_rms(soundvalue)
+                self.meter(soundvalue)
                 if self.action:
-                    if self.is_triggered(rms):
-                        self.execute(rms)
-                self.monitor(rms)
+                    if self.is_triggered(soundvalue):
+                        self.execute(soundvalue)
+                self.monitor(soundvalue)
             self.is_running = False
             self.stop()
 
         except self.__class__.StopException:
             self.is_running = False
             self.stop()
 
-    def meter(self, rms):
+    def meter(self, value):
         if not self._graceful:
-            sys.stdout.write('\r%10d  ' % rms)
+            sys.stdout.write('\r%10d  %s' % (value, self.valuestype))
             sys.stdout.flush()
             if self.log:
-                self.logging.info(rms)
+                self.logging.info(value)
+
+    def weightedvalue(self, segment, curve):
+        """
+        This function it is used to calculate the average dB level using the weighting curves.
+        The zeros and poles were created by using those functions:
+        http://siggigue.github.io/pyfilterbank/splweighting.html
+        with the sample rate hardcoded to 44100Hz
+        More info about weighting curves:
+        https://www.cirrusresearch.co.uk/blog/2011/08/what-are-a-c-z-frequency-weightings/
+        :param segment: The segment to calculate the weighted average
+        :param curve: The type of curve to apply Z, A, B or C
+        :return: The weighted average
+        """
+        if curve.lower() == "z":
+            return segment.dBFS
+        if curve.lower() == "a":
+            b = np.array([0.25574113, -0.51148225, -0.25574113, 1.0229645, -0.25574113,
```

Comment:
@jajaxdlol Are these hardcoded numbers only for the 44100 frequency?

Reply:
Those are hardcoded for 44100 freq yes. I'm not good at filter design so I have no idea on how to create a function to create curves according to frequency, I wasn't sure if you wanted to add a dependency to another library.